### ISSUE TITLE
Serve over HTTPS

### ISSUE THREAD

**Issue by acme** [fastly/Viceroy#50]
👋 Have we considered serving `fastly compute serve` over HTTPS instead of HTTP? My application isn’t working because I set HTTPS-only cookies, and I don’t want to make too many changes to the application so that it works locally.

Here's a screenshot from Google Chrome developer tools showing a warning about the `Set-Cookie` header:

<img width="1181" alt="Screenshot_2021-07-20_at_10_18_09" src="https://user-images.githubusercontent.com/19988/126288320-7cf3db01-2eb7-458a-8f7b-f664c466effe.png">

**Comment by fgsch** [comment]
This might be another form of #12 .

**Comment by mgattozzi** [comment]
It might be related, but I don't think this is necessarily the same. From the README the limitation #12 is trying to address is that "Information about the TLS connection from the client is not available". This issue could be implemented without exposing that information to a guest. If I'm understanding #12 it's more for functions in our SDK like [`get_tls_client_hello`](https://docs.rs/fastly/0.7.3/fastly/http/struct.Request.html#method.get_tls_client_hello) which in turn make [hostcalls](https://docs.rs/fastly/0.7.3/src/fastly/http/request/handle.rs.html#840) that as we can see [here in Viceroy](https://github.com/fastly/Viceroy/blob/main/lib/src/wiggle_abi/req_impl.rs#L106-L114) are not handled yet. Implementing this would be a necessary first step to exposing this information to Viceroy users.

**Comment by fgsch** [comment]
@mgattozzi Yup, totally. What I meant in my rather terse comment (sorry!) is that comparatively speaking #12 is quite small, so implementing the foundations for #12 will also cover this, but you are 100% correct and I appreciate the clarification 👍

**Comment by mgattozzi** [comment]
Ah I see! No worries @fgsch 😄 Was adding some clarification, especially for those who might not be familiar with the context!
